Maven Build Issue - Test Failure

I am having an issue building the jar with Maven. I have the latest code from master. When I run mvn package I get the following test error

```
[ERROR] Tests run: 11, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.995 s <<< FAILURE! - in net.juniper.netconf.NetconfSessionTest
[ERROR] GIVEN_executeRPC_WHEN_lldpRequest_THEN_correctResponse(net.juniper.netconf.NetconfSessionTest)  Time elapsed: 1.567 s  <<< FAILURE!
org.junit.ComparisonFailure:
expected:<...nos" message-id="3">[
    <lldp-neighbors-information junos:style="brief">
        <lldp-neighbor-information>
            <lldp-local-port-id>xe-0/0/9:1</lldp-local-port-id>
            <lldp-local-parent-interface-name>ae51</lldp-local-parent-interface-name>
            <lldp-remote-chassis-id-subtype>Mac address</lldp-remote-chassis-id-subtype>
            <lldp-remote-chassis-id>00:00:00:80:1e:00</lldp-remote-chassis-id>
            <lldp-remote-port-description>ethernet  6/1</lldp-remote-port-description>
            <lldp-remote-system-name>test.system.new</lldp-remote-system-name>
        </lldp-neighbor-information>
]   </lldp-neighbors-information>
</rpc-reply>
> but was:<...nos" message-id="3">[
    <lldp-neighbors-information junos:style="brief">
        <lldp-neighbor-information>
            <lldp-local-port-id>xe-0/0/9:1</lldp-local-port-id>
            <lldp-local-parent-interface-name>ae51</lldp-local-parent-interface-name>
            <lldp-remote-chassis-id-subtype>Mac address</lldp-remote-chassis-id-subtype>
            <lldp-remote-chassis-id>00:00:00:80:1e:00</lldp-remote-chassis-id>
            <lldp-remote-port-description>ethernet  6/1</lldp-remote-port-description>
            <lldp-remote-system-name>test.system.new</lldp-remote-system-name>
        </lldp-neighbor-information>
    </lldp-neighbors-information>]
</rpc-reply>
>
        at net.juniper.netconf.NetconfSessionTest.GIVEN_executeRPC_WHEN_lldpRequest_THEN_correctResponse(NetconfSessionTest.java:182)

[INFO] Running net.juniper.netconf.XMLTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.012 s - in net.juniper.netconf.XMLTest
[INFO] 
[INFO] Results:
[INFO]
[ERROR] Failures: 
[ERROR]   NetconfSessionTest.GIVEN_executeRPC_WHEN_lldpRequest_THEN_correctResponse:182 expected:<...nos" message-id="3">[
    <lldp-neighbors-information junos:style="brief">
        <lldp-neighbor-information>
            <lldp-local-port-id>xe-0/0/9:1</lldp-local-port-id>
            <lldp-local-parent-interface-name>ae51</lldp-local-parent-interface-name>
            <lldp-remote-chassis-id-subtype>Mac address</lldp-remote-chassis-id-subtype>
            <lldp-remote-chassis-id>00:00:00:80:1e:00</lldp-remote-chassis-id>
            <lldp-remote-port-description>ethernet  6/1</lldp-remote-port-description>
            <lldp-remote-system-name>test.system.new</lldp-remote-system-name>
        </lldp-neighbor-information>
]   </lldp-neighbors-information>
</rpc-reply>
> but was:<...nos" message-id="3">[
    <lldp-neighbors-information junos:style="brief">
        <lldp-neighbor-information>
            <lldp-local-port-id>xe-0/0/9:1</lldp-local-port-id>
            <lldp-local-parent-interface-name>ae51</lldp-local-parent-interface-name>
            <lldp-remote-chassis-id-subtype>Mac address</lldp-remote-chassis-id-subtype>
            <lldp-remote-chassis-id>00:00:00:80:1e:00</lldp-remote-chassis-id>
            <lldp-remote-port-description>ethernet  6/1</lldp-remote-port-description>
            <lldp-remote-system-name>test.system.new</lldp-remote-system-name>
        </lldp-neighbor-information>
    </lldp-neighbors-information>]
</rpc-reply>
>
[INFO]
[ERROR] Tests run: 21, Failures: 1, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 6.642 s
[INFO] Finished at: 2021-02-09T14:21:37-05:00
[INFO] ------------------------------------------------------------------------


```

Getting this same error.

I am using Windows 10.
Java :  jdk1.8.0_271



>
        at net.juniper.netconf.NetconfSessionTest.GIVEN_executeRPC_WHEN_lldpRequest_THEN_correctResponse(NetconfSessionTest.java:182)

[INFO] Running net.juniper.netconf.XMLTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.012 s - in net.juniper.netconf.XMLTest
[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   NetconfSessionTest.GIVEN_executeRPC_WHEN_lldpRequest_THEN_correctResponse:182 expected:<...nos" message-id="3">[
    <lldp-neighbors-information junos:style="brief">
        <lldp-neighbor-information>
            <lldp-local-port-id>xe-0/0/9:1</lldp-local-port-id>
            <lldp-local-parent-interface-name>ae51</lldp-local-parent-interface-name>
            <lldp-remote-chassis-id-subtype>Mac address</lldp-remote-chassis-id-subtype>
            <lldp-remote-chassis-id>00:00:00:80:1e:00</lldp-remote-chassis-id>
            <lldp-remote-port-description>ethernet  6/1</lldp-remote-port-description>
            <lldp-remote-system-name>test.system.new</lldp-remote-system-name>
        </lldp-neighbor-information>
]   </lldp-neighbors-information>
</rpc-reply>
> but was:<...nos" message-id="3">[
    <lldp-neighbors-information junos:style="brief">
        <lldp-neighbor-information>
            <lldp-local-port-id>xe-0/0/9:1</lldp-local-port-id>
            <lldp-local-parent-interface-name>ae51</lldp-local-parent-interface-name>
            <lldp-remote-chassis-id-subtype>Mac address</lldp-remote-chassis-id-subtype>
            <lldp-remote-chassis-id>00:00:00:80:1e:00</lldp-remote-chassis-id>
            <lldp-remote-port-description>ethernet  6/1</lldp-remote-port-description>
            <lldp-remote-system-name>test.system.new</lldp-remote-system-name>
        </lldp-neighbor-information>
    </lldp-neighbors-information>]
</rpc-reply>
>
[INFO]
[ERROR] Tests run: 21, Failures: 1, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.623 s
[INFO] Finished at: 2021-05-28T12:58:09+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test (default-test) on project netconf-java: There are test failures.
[ERROR]
[ERROR] Please refer to C:\Users\932478\Juniper-NetConf-java\netconf-java\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[error.md](https://github.com/Juniper/netconf-java/files/6559104/error.md)



I am using Windows 10.
Java :  jdk1.8.0_271

--- a/pom.xml
+++ b/pom.xml
@@ -7,7 +7,7 @@
 
     <groupId>net.juniper.netconf</groupId>
     <artifactId>netconf-java</artifactId>
-    <version>2.1.1.2</version>
+    <version>2.1.1.3</version>
     <packaging>jar</packaging>
 
     <properties>
@@ -151,7 +151,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
-                <version>3.1.0</version>
+                <version>3.2.0</version>
                 <configuration>
                     <show>public</show>
                 </configuration>
@@ -229,7 +229,7 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>r05</version>
+            <version>30.0-jre</version>
         </dependency>
 
     </dependencies>

--- a/src/main/java/net/juniper/netconf/NetconfConstants.java
+++ b/src/main/java/net/juniper/netconf/NetconfConstants.java
@@ -33,5 +33,6 @@ public class NetconfConstants {
 
     public static final String EMPTY_LINE = "";
     public static final String LF = "\n";
+    public static final String CR = "\r";
 
 }

--- a/src/main/java/net/juniper/netconf/XML.java
+++ b/src/main/java/net/juniper/netconf/XML.java
@@ -391,7 +391,7 @@ public class XML {
                     return null;
                 }
             }        }
-        if (nextElement == null) {
+        if (nextElement == null || nextElement.getFirstChild() == null || nextElement.getFirstChild().getNodeValue()== null) {
             return null;
         }
         String value = nextElement.getFirstChild().getNodeValue();

--- a/src/test/java/net/juniper/netconf/NetconfSessionTest.java
+++ b/src/test/java/net/juniper/netconf/NetconfSessionTest.java
@@ -1,5 +1,6 @@
 package net.juniper.netconf;
 
+import com.google.common.base.Charsets;
 import com.jcraft.jsch.Channel;
 import lombok.extern.slf4j.Slf4j;
 import org.junit.Before;
@@ -33,10 +34,6 @@ public class NetconfSessionTest {
     public static final int COMMAND_TIMEOUT = 5000;
 
     private static final String FAKE_HELLO = "fake hello";
-    private static final String MERGE_LOAD_TYPE = "merge";
-    private static final String REPLACE_LOAD_TYPE = "replace";
-    private static final String BAD_LOAD_TYPE = "other";
-    private static final String FAKE_TEXT_FILE_PATH = "fakepath";
 
     private static final String DEVICE_PROMPT = "]]>]]>";
     private static final byte[] DEVICE_PROMPT_BYTE = DEVICE_PROMPT.getBytes();
@@ -88,7 +85,7 @@ public class NetconfSessionTest {
     }
 
     @Test
-    public void GIVEN_createSession_WHEN_timeoutExceeded_THEN_throwSocketTimeoutException() throws Exception {
+    public void GIVEN_createSession_WHEN_timeoutExceeded_THEN_throwSocketTimeoutException() {
         Thread thread = new Thread(() -> {
             try {
                 outPipe.write(FAKE_RPC_REPLY.getBytes());
@@ -111,7 +108,7 @@ public class NetconfSessionTest {
     }
 
     @Test
-    public void GIVEN_createSession_WHEN_connectionClose_THEN_throwSocketTimeoutException() throws Exception {
+    public void GIVEN_createSession_WHEN_connectionClose_THEN_throwSocketTimeoutException() {
         Thread thread = new Thread(() -> {
             try {
                 outPipe.write(FAKE_RPC_REPLY.getBytes());
@@ -155,6 +152,8 @@ public class NetconfSessionTest {
     @Test
     public void GIVEN_executeRPC_WHEN_lldpRequest_THEN_correctResponse() throws Exception {
         byte[] lldpResponse = Files.readAllBytes(TestHelper.getSampleFile("responses/lldpResponse.xml").toPath());
+        String expectedResponse = new String(lldpResponse, Charsets.UTF_8)
+                .replaceAll(NetconfConstants.CR, NetconfConstants.EMPTY_LINE) + NetconfConstants.LF;
 
         Thread thread = new Thread(() -> {
             try {
@@ -179,7 +178,7 @@ public class NetconfSessionTest {
         Thread.sleep(200);
         String deviceResponse = netconfSession.executeRPC(TestConstants.LLDP_REQUEST).toString();
 
-        assertEquals(new String(lldpResponse) + NetconfConstants.LF, deviceResponse);
+        assertEquals(expectedResponse, deviceResponse);
     }
 
     @Test
@@ -187,7 +186,7 @@ public class NetconfSessionTest {
         when(mockNetconfSession.executeRPC(eq(TestConstants.LLDP_REQUEST))).thenCallRealMethod();
         when(mockNetconfSession.getRpcReply(anyString())).thenReturn(NETCONF_SYNTAX_ERROR_MSG_FROM_DEVICE);
 
-        assertThatThrownBy(() -> mockNetconfSession.executeRPC(TestConstants.LLDP_REQUEST).toString())
+        assertThatThrownBy(() -> mockNetconfSession.executeRPC(TestConstants.LLDP_REQUEST))
                 .isInstanceOf(NetconfException.class)
                 .hasMessage("Netconf server detected an error: netconf error: syntax error");
     }

